R/calcHDDCDD.R

```diff
@@ -1,101 +1,714 @@
+                   f[f$variable == "tas"       & f$gcm == m,][[n, "file"]])
+
+      if (bait){
+        frsds <- gsub(".nc",
+                      paste0("_", suffix, ".nc"),
+                      f[f$variable == "rsds"    & f$gcm == m,][[n, "file"]])
+
+        fsfc  <- gsub(".nc",
+                      paste0("_", suffix, ".nc"),
+                      f[f$variable == "sfcwind" & f$gcm == m,][[n, "file"]])
+
+        fhuss <- gsub(".nc",
+                      paste0("_", suffix, ".nc"),
+                      f[f$variable == "huss"    & f$gcm == m,][[n, "file"]])
+      }
+    }
+
+    else {
+      ftas  <- f[f$variable == "tas"     & f$gcm == m,][[n, "file"]]
+      frsds <- f[f$variable == "rsds"    & f$gcm == m,][[n, "file"]]
+      fsfc  <- f[f$variable == "sfcwind" & f$gcm == m,][[n, "file"]]
+      fhuss <- f[f$variable == "huss"    & f$gcm == m,][[n, "file"]]
+    }
+
+    print(paste("Processing temperature file:", ftas))
+
+    if(bait) {
+      hddcddCell <- calcStackHDDCDD(ftas,
+                                    t_lim,
+                                    countries,
+                                    pop,
+                                    hddcddFactor,
+                                    bait,
+                                    frsds = frsds,
+                                    fsfc  = fsfc,
+                                    fhuss = fhuss,
+                                    wBAIT = wBAIT,
+                                    params = params)}
+
+    else {
+      hddcddCell <- calcStackHDDCDD(ftas,
+                                    t_lim,
+                                    countries,
+                                    pop,
+                                    hddcddFactor,
+                                    bait)}
+    return(hddcddCell)
+  }
+
+
+  plotRaster <- function(r) {
+    library("ggplot2")
+    r_map <- stack(as.data.frame(raster::getValues(r)))
+    coords <- raster::xyFromCell(r, seq_len(raster::ncell(r)))
+    names(r_map) <- c('value', 'variable')
+    r_map <- cbind(coords, r_map)
+    ggplot(r_map) +
+      geom_tile(aes(x, y, fill = value)) +
+      facet_wrap(~ variable) +
+      scale_fill_gradientn(colours = rev(terrain.colors(225))) +
+      coord_equal()
+  }
+
+
+
+  # PARAMETERS------------------------------------------------------------------
+
+  terra::setGDALconfig(c("BIGTIFF = YES"))
+
+
+  # threshold temperature for heating and cooling [C]
+  # NOTE: Staffel gives global average of T_heat = 14, T_cool = 20
+  # t_lim <- list("HDD" = seq(12, 18), "CDD" = seq(20, 26))
+  t_lim <- list("HDD" = seq(14), "CDD" = seq(22))
+
+  # standard deviations for temperature distributions
+  tlim_std <- 5   # threshold
+  tamb_std <- 5   # ambient
+
+  # historical years    -> NOTE: not sure if further needed
+  firstHistYear <- 1950
+  lastHistYear  <- 2009
+
+  # range of pre-calculated HDD/CDD-values, e.g. [223, 232] K, converted to [C]
+  tlow <- 223 - 273.15
+  tup <- 323 - 273.15
```

Comment:
Is this left-over code that is not used? Cause 223K do not exist.

Reply:
Why would 223K not exist? This was code that was supposed to define the lower/upper boundaries for calcHDDCDDFactors which I have not incorporated yet.

Reply:
You're right. No idea what I was talking about. Sorry for that.